.github/actions/frontend-plugin-cache/action.yml

```diff
@@ -0,0 +1,26 @@
+name: Frontend Plugin Cache
+description: Caches Node.js installation for the frontend-maven-plugin
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/cache@v3
+      name: Cache Node.js installation
+      with:
+        path: |
+          node
+        key: ${{ runner.os }}-frontend-plugin-node-${{ hashFiles('pom.xml') }}
```

Comment:
For the maven-cache action, we're using a weekly cache key with no restore-key.

This avoids too many caches, as `pom.xml`s change frequently, and restore-keys can make the caches grow with time. 

Would you consider changing this to weekly caches as well?

With the Maven cache, we eventually added a `create-cache-if-it-doesnt-exist` parameter, as we would want to cache only the artifacts created in the build, and the re-use them in other workflows, as the other workflows download fewer dependencies, and would then upload only a subset of the information we want to cache.

Reply:
In this case it was only referencing the root POM, but I am really only interested in change to the Node.js and PNPM versions. I've changed this so the cache is only restored if `node.version` and `pnpm.version` match. This should also remove the need to keep weekly caches, as it's much more fine grained already.

> With the Maven cache, we eventually added a `create-cache-if-it-doesnt-exist` parameter, as we would want to cache only the artifacts created in the build, and the re-use them in other workflows, as the other workflows download fewer dependencies, and would then upload only a subset of the information we want to cache.

Mhhh, yeah I wonder if this needs/should be done for PNPM dependencies. We can always optimize this later if it causes trouble.

Reply:
I'll consider this one resolved.